src/common/base/Configuration.cpp

```diff
@@ -160,7 +164,7 @@ Status Configuration::fetchAsSubConf(const char *key, Configuration &subconf) co
 }
 
 
-Status Configuration::updateStringField(const char* key, const std::string& val) {
+Status Configuration::upsertStringField(const char* key, const std::string& val) {
```

Comment:
A question, Why naming `upsert`?
What I understand that upsert is : should be override the old value  if key exists;  should be insert  a new value if the key doesn't exist.

Reply:
Exactly, so we can insert some option which doesn't exists, and update some option with a new value. The function actually is a upsert, a bad name.